src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncDispatcher.cs

```diff
@@ -151,12 +151,13 @@ public async Task Start(CancellationToken cancellationToken)
 
             if (Feed.IsMultiFeed)
             {
-                // Don't free if num of concurrent processing threads is limited
-                // Otherwise, it'll keep downloading more requests.
+                // Limit multithreaded feed concurrency. Note, this also blocks freeing the allocation, which is deliberate.
+                // otherwise, we will keep spawning requests without processing it fast enough, which consume memory.
                 await _concurrentProcessingSemaphore.WaitAsync(cancellationToken);
-                Free(allocation);
             }
 
+            Free(allocation);
```

Comment:
Should it be in finally close of the above try?

Reply:
Why? It does the same thing. Must be after the semaphore though.

Reply:
So on exception we will never release peer?

Reply:
These is a catch all in line 147. This always execute.